pkg/goldpinger/client.go

```diff
@@ -54,6 +55,67 @@ func CheckNeighboursNeighbours(ctx context.Context) *models.CheckAllResults {
 	return CheckAllPods(ctx, SelectPods())
 }
 
+// CheckCluster does a CheckNeighboursNeighbours and analyses results to produce a binary OK or not OK
+func CheckCluster(ctx context.Context) *models.ClusterHealthResults {
+
+	start := time.Now()
+	response := models.ClusterHealthResults{
+		GeneratedAt: strfmt.DateTime(start),
+		OK:          true,
+	}
+	selectedPods := SelectPods()
+
+	// precompute the expected set of nodes
+	expectedNodes := []string{}
+	for _, peer := range selectedPods {
+		expectedNodes = append(expectedNodes, peer.HostIP)
+	}
+	sort.Strings(expectedNodes)
+
+	// get the response we serve for check_all
+	checkAll := CheckAllPods(ctx, selectedPods)
+
+	// we should at the very least have a response from ourselves
+	if len(checkAll.Responses) < 1 {
+		response.OK = false
+	}
+	for _, resp := range checkAll.Responses {
```

Comment:
Having `resp` and `response` is a little confusing. I would change resp to `peerResp` or something similar... (Sorry golang Gods, I know you love short variable names, perhaps `pr` would be better?)